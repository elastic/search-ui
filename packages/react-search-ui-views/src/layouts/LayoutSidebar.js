import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "../view-helpers";

class LayoutSidebar extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node
  };

  constructor(props) {
    super(props);
    this.state = { isSidebarToggled: false };
  }

  toggleSidebar = () => {
    this.setState(({ isSidebarToggled }) => ({
      isSidebarToggled: !isSidebarToggled
    }));
  };

  renderToggleButton = label => {
    if (!this.props.children) return null;

    return (
      <button
        hidden
        type="button"
        className="sui-layout-sidebar-toggle"
        onClick={this.toggleSidebar}
      >
        {label}
      </button>
    );
  };

  render() {
    const { className, children } = this.props;
    const { isSidebarToggled } = this.state;

    const classes = appendClassName(
      className,
      isSidebarToggled ? `${className}--toggled` : null
    );

    return (
      <>
        {this.renderToggleButton("Show Filters")}
        <div className={classes}>
          {this.renderToggleButton("Save Filters")}
          {children}
        </div>
      </>
    );
  }
}

export default LayoutSidebar;
