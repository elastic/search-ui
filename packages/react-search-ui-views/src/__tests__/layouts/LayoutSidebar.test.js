import React from "react";
import { shallow } from "enzyme";

import LayoutSidebar from "../../layouts/LayoutSidebar";

it("renders correctly", () => {
  const wrapper = shallow(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders toggled class based on state", () => {
  const wrapper = shallow(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );
  wrapper.setState({ isSidebarToggled: true });

  expect(wrapper.find(".sui-layout-sidebar--toggled")).toHaveLength(1);
});

it("updates isSidebarToggled state on button click", () => {
  const wrapper = shallow(
    <LayoutSidebar className="sui-layout-sidebar">Hello world!</LayoutSidebar>
  );
  expect(wrapper.state("isSidebarToggled")).toEqual(false);
  const buttons = wrapper.find(".sui-layout-sidebar-toggle");

  buttons.first().simulate("click");
  expect(wrapper.state("isSidebarToggled")).toEqual(true);

  buttons.last().simulate("click");
  expect(wrapper.state("isSidebarToggled")).toEqual(false);
});
