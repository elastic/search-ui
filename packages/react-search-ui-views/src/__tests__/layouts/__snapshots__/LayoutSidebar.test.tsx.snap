// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <button
    className="sui-layout-sidebar-toggle"
    hidden={true}
    onClick={[Function]}
    type="button"
  >
    Show Filters
  </button>
  <div
    className="sui-layout-sidebar"
  >
    <button
      className="sui-layout-sidebar-toggle"
      hidden={true}
      onClick={[Function]}
      type="button"
    >
      Save Filters
    </button>
    Hello world!
  </div>
</Fragment>
`;
